test/spec/controllers/challengesCtrlSpec.js

```diff
@@ -184,60 +184,78 @@ describe('Challenges Controller', function() {
-        }
-      }
-      expect( scope.newChallenge.dailys ).to.eql( copyChallenge.dailys );
-      expect( scope.newChallenge.todos ).to.eql( copyChallenge.todos );
-      expect( scope.newChallenge.rewards ).to.eql( copyChallenge.rewards );
-      expect( scope.newChallenge.leader ).to.eql( copyChallenge.leader );
-      expect( scope.newChallenge.timestamp ).to.not.eql( copyChallenge.timestamp );
-      expect( scope.newChallenge.members ).to.eql( [] );
-      expect( scope.newChallenge.official ).to.eql( copyChallenge.official );
-      expect( scope.newChallenge.prize ).to.eql( copyChallenge.prize );
+      scope.clone(challengeToClone);
 
+      expect(scope.newChallenge.name).to.eql(challengeToClone.name);
+      expect(scope.newChallenge.shortName).to.eql(challengeToClone.shortName);
+      expect(scope.newChallenge.description).to.eql(challengeToClone.description);
+      expect(scope.newChallenge.leader).to.eql(user._id);
+      expect(scope.newChallenge.group).to.eql(challengeToClone.group._id);
+      expect(scope.newChallenge.official).to.eql(challengeToClone.official);
+      expect(scope.newChallenge.prize).to.eql(challengeToClone.prize);
     });
 
-  });
+    it('does not clone members', function() {
+      scope.clone(challengeToClone);
 
+      expect(scope.newChallenge.members).to.not.exist;
+    });
+
+    it('does not clone timestamp', function() {
+      scope.clone(challengeToClone);
+
+      expect(scope.newChallenge.timestamp).to.not.exist;
+    });
+
+    it('clones habits', function() {
+      scope.clone(challengeToClone);
+
+      expect(scope.newChallenge.habits.length).to.eql(challengeToClone.habits.length);
+      expect(scope.newChallenge.habits[0].text).to.eql(challengeToClone.habits[0].text);
+      expect(scope.newChallenge.habits[0].notes).to.eql(challengeToClone.habits[0].notes);
+    });
+
+    it('clones dailys', function() {
+      scope.clone(challengeToClone);
+
+      expect(scope.newChallenge.dailys.length).to.eql(challengeToClone.dailys.length);
+      expect(scope.newChallenge.dailys[0].text).to.eql(challengeToClone.dailys[0].text);
+      expect(scope.newChallenge.dailys[0].notes).to.eql(challengeToClone.dailys[0].notes);
+    });
+
+    it('clones todos', function() {
+      scope.clone(challengeToClone);
+
+      expect(scope.newChallenge.todos.length).to.eql(challengeToClone.todos.length);
+      expect(scope.newChallenge.todos[0].text).to.eql(challengeToClone.todos[0].text);
+      expect(scope.newChallenge.todos[0].notes).to.eql(challengeToClone.todos[0].notes);
+    });
+
+    it('clones rewards', function() {
+      scope.clone(challengeToClone);
+
+      expect(scope.newChallenge.rewards.length).to.eql(challengeToClone.rewards.length);
+      expect(scope.newChallenge.rewards[0].text).to.eql(challengeToClone.rewards[0].text);
+      expect(scope.newChallenge.rewards[0].notes).to.eql(challengeToClone.rewards[0].notes);
```

Comment:
I ended up testing the nitty gritty of the task details in the Tasks.cloneTask function